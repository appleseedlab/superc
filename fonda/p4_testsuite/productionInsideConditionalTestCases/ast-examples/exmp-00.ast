Processing Exp-Header-Action.p4 ...
TranslationUnit(
  start(
    program(
      input(
        Conditional(
          (declare-fun |(defined CONFIG_B)| () Bool)(declare-fun |(defined CONFIG_A)| () Bool)(assert (or (and |(defined CONFIG_A)| |(defined CONFIG_B)|)    (and |(defined CONFIG_A)| (not |(defined CONFIG_B)|)))),
          declaration(
            typeDeclaration(
              derivedTypeDeclaration(
                headerTypeDeclaration(
                  optAnnotations(),
                  superc.core.Syntax$Language(
                    "header"
                  ),
                  name(
                    nonTypeName(
                      superc.core.Syntax$Text(
                        "int_data_t"
                      )
                    )
                  ),
                  optTypeParameters(),
                  superc.core.Syntax$Language(
                    "{"
                  ),
                  structFieldList(
                    Conditional(
                      (declare-fun |(defined CONFIG_A)| () Bool)(assert |(defined CONFIG_A)|),
                      structField(
                        optAnnotations(),
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "varbit"
                            ),
                            l_angle(
                              superc.core.Syntax$Language(
                                "<"
                              )
                            ),
                            superc.core.Syntax$Text(
                              "8032"
                            ),
                            r_angle(
                              superc.core.Syntax$Language(
                                ">"
                              )
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "data"
                            )
                          )
                        ),
                        superc.core.Syntax$Language(
                          ";"
                        )
                      )
                    )
                  ),
                  superc.core.Syntax$Language(
                    "}"
                  )
                )
              )
            )
          )
        ),
        Conditional(
          (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
          declaration(
            actionDeclaration(
              Conditional(
                (declare-fun |(defined CONFIG_B)| () Bool)(declare-fun |(defined CONFIG_A)| () Bool)(assert (let ((a!1 (or (and (not |(defined CONFIG_A)|) |(defined CONFIG_B)|)               (and (not |(defined CONFIG_A)|) (not |(defined CONFIG_B)|)))))  (or (and a!1 |(defined CONFIG_B)|) (and a!1 (not |(defined CONFIG_B)|))))),
                optAnnotations(),
                (declare-fun |(defined CONFIG_B)| () Bool)(declare-fun |(defined CONFIG_A)| () Bool)(assert (or (and |(defined CONFIG_A)| |(defined CONFIG_B)|)    (and |(defined CONFIG_A)| (not |(defined CONFIG_B)|)))),
                optAnnotations()
              ),
              superc.core.Syntax$Language(
                "action"
              ),
              name(
                nonTypeName(
                  superc.core.Syntax$Text(
                    "ac1"
                  )
                )
              ),
              superc.core.Syntax$Language(
                "("
              ),
              parameterList(
                Conditional(
                  (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                  nonEmptyParameterList(
                    Conditional(
                      (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                      parameter(
                        optAnnotations(),
                        direction(),
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "int"
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "j"
                            )
                          )
                        )
                      )
                    ),
                    Conditional(
                      (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                      superc.core.Syntax$Language(
                        ","
                      )
                    ),
                    Conditional(
                      (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                      parameter(
                        optAnnotations(),
                        direction(),
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "int"
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "i"
                            )
                          )
                        )
                      )
                    )
                  )
                )
              ),
              superc.core.Syntax$Language(
                ")"
              ),
              blockStatement(
                optAnnotations(),
                superc.core.Syntax$Language(
                  "{"
                ),
                statOrDeclList(
                  Conditional(
                    (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                    statementOrDeclaration(
                      variableDeclaration(
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "int"
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "j"
                            )
                          )
                        ),
                        optInitializer(
                          superc.core.Syntax$Language(
                            "="
                          ),
                          initializer(
                            expression(
                              Conditional(
                                (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                                superc.core.Syntax$Text(
                                  "5"
                                )
                              )
                            )
                          )
                        ),
                        superc.core.Syntax$Language(
                          ";"
                        )
                      )
                    )
                  )
                ),
                superc.core.Syntax$Language(
                  "}"
                )
              )
            )
          )
        ),
        Conditional(
          ,
          declaration(
            actionDeclaration(
              Conditional(
                (declare-fun |(defined CONFIG_B)| () Bool)(assert (not |(defined CONFIG_B)|)),
                Conditional(
                  (declare-fun |(defined CONFIG_B)| () Bool)(declare-fun |(defined CONFIG_A)| () Bool)(assert (let ((a!1 (or (and (not |(defined CONFIG_A)|) |(defined CONFIG_B)|)               (and (not |(defined CONFIG_A)|) (not |(defined CONFIG_B)|)))))  (or (and a!1 |(defined CONFIG_B)|) (and a!1 (not |(defined CONFIG_B)|))))),
                  optAnnotations(),
                  (declare-fun |(defined CONFIG_B)| () Bool)(declare-fun |(defined CONFIG_A)| () Bool)(assert (or (and |(defined CONFIG_A)| |(defined CONFIG_B)|)    (and |(defined CONFIG_A)| (not |(defined CONFIG_B)|)))),
                  optAnnotations()
                ),
                (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                optAnnotations()
              ),
              superc.core.Syntax$Language(
                "action"
              ),
              Conditional(
                (declare-fun |(defined CONFIG_B)| () Bool)(assert (not |(defined CONFIG_B)|)),
                name(
                  nonTypeName(
                    superc.core.Syntax$Text(
                      "ac2"
                    )
                  )
                ),
                (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                name(
                  nonTypeName(
                    superc.core.Syntax$Text(
                      "ac2"
                    )
                  )
                )
              ),
              superc.core.Syntax$Language(
                "("
              ),
              Conditional(
                (declare-fun |(defined CONFIG_B)| () Bool)(assert (not |(defined CONFIG_B)|)),
                parameterList(),
                (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                parameterList()
              ),
              superc.core.Syntax$Language(
                ")"
              ),
              blockStatement(
                optAnnotations(),
                superc.core.Syntax$Language(
                  "{"
                ),
                statOrDeclList(
                  Conditional(
                    ,
                    statementOrDeclaration(
                      variableDeclaration(
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "int"
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "j"
                            )
                          )
                        ),
                        optInitializer(
                          superc.core.Syntax$Language(
                            "="
                          ),
                          initializer(
                            expression(
                              Conditional(
                                ,
                                superc.core.Syntax$Text(
                                  "5"
                                )
                              )
                            )
                          )
                        ),
                        superc.core.Syntax$Language(
                          ";"
                        )
                      )
                    )
                  )
                ),
                superc.core.Syntax$Language(
                  "}"
                )
              )
            )
          )
        )
      )
    )
  )
)
