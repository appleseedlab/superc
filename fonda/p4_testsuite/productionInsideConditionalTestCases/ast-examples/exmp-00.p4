#ifdef CONFIG_A
header int_data_t {
    varbit<8032> data;
}
#endif

#ifdef CONFIG_B
action ac1 (int j, int i) {
    int j = 5;
}
#endif


action ac2 () {
    int j = 5;
}