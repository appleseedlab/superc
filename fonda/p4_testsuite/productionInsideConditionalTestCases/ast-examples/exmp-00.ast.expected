Processing Exp-Header-Action.p4 ...
TranslationUnit(
  start(
    program(
      input(
        Conditional(
          (declare-fun |(defined CONFIG_A)| () Bool)(assert |(defined CONFIG_A)|),
          declaration(
            typeDeclaration(
              derivedTypeDeclaration(
                headerTypeDeclaration(
                  optAnnotations(),
                  superc.core.Syntax$Language(
                    "header"
                  ),
                  name(
                    nonTypeName(
                      superc.core.Syntax$Text(
                        "int_data_t"
                      )
                    )
                  ),
                  optTypeParameters(),
                  superc.core.Syntax$Language(
                    "{"
                  ),
                  structFieldList(
                    Conditional(
                      (declare-fun |(defined CONFIG_A)| () Bool)(assert |(defined CONFIG_A)|),
                      structField(
                        optAnnotations(),
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "varbit"
                            ),
                            l_angle(
                              superc.core.Syntax$Language(
                                "<"
                              )
                            ),
                            superc.core.Syntax$Text(
                              "8032"
                            ),
                            r_angle(
                              superc.core.Syntax$Language(
                                ">"
                              )
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "data"
                            )
                          )
                        ),
                        superc.core.Syntax$Language(
                          ";"
                        )
                      )
                    )
                  ),
                  superc.core.Syntax$Language(
                    "}"
                  )
                )
              )
            )
          )
        ),
        Conditional(
          (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
          declaration(
            actionDeclaration(
              Conditional(
                (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                optAnnotations(),
              ),
              superc.core.Syntax$Language(
                "action"
              ),
              name(
                nonTypeName(
                  superc.core.Syntax$Text(
                    "ac1"
                  )
                )
              ),
              superc.core.Syntax$Language(
                "("
              ),
              parameterList(
                Conditional(
                  (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                  nonEmptyParameterList(
                    Conditional(
                      (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                      parameter(
                        optAnnotations(),
                        direction(),
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "int"
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "j"
                            )
                          )
                        )
                      )
                    ),
                    Conditional(
                      (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                      superc.core.Syntax$Language(
                        ","
                      )
                    ),
                    Conditional(
                      (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                      parameter(
                        optAnnotations(),
                        direction(),
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "int"
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "i"
                            )
                          )
                        )
                      )
                    )
                  )
                )
              ),
              superc.core.Syntax$Language(
                ")"
              ),
              blockStatement(
                optAnnotations(),
                superc.core.Syntax$Language(
                  "{"
                ),
                statOrDeclList(
                  Conditional(
                    (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                    statementOrDeclaration(
                      variableDeclaration(
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "int"
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "j"
                            )
                          )
                        ),
                        optInitializer(
                          superc.core.Syntax$Language(
                            "="
                          ),
                          initializer(
                            expression(
                              Conditional(
                                (declare-fun |(defined CONFIG_B)| () Bool)(assert |(defined CONFIG_B)|),
                                superc.core.Syntax$Text(
                                  "5"
                                )
                              )
                            )
                          )
                        ),
                        superc.core.Syntax$Language(
                          ";"
                        )
                      )
                    )
                  )
                ),
                superc.core.Syntax$Language(
                  "}"
                )
              )
            )
          )
        ),
        Conditional(
          ,
          declaration(
            actionDeclaration(
              Conditional(
                ,
              ),
              superc.core.Syntax$Language(
                "action"
              ),
              Conditional(
                ,
                name(
                  nonTypeName(
                    superc.core.Syntax$Text(
                      "ac2"
                    )
                  )
                ),
                name(
                  nonTypeName(
                    superc.core.Syntax$Text(
                      "ac2"
                    )
                  )
                )
              ),
              superc.core.Syntax$Language(
                "("
              ),
              Conditional(
                parameterList()
              ),
              superc.core.Syntax$Language(
                ")"
              ),
              blockStatement(
                optAnnotations(),
                superc.core.Syntax$Language(
                  "{"
                ),
                statOrDeclList(
                  Conditional(
                    ,
                    statementOrDeclaration(
                      variableDeclaration(
                        typeRef(
                          baseType(
                            superc.core.Syntax$Language(
                              "int"
                            )
                          )
                        ),
                        name(
                          nonTypeName(
                            superc.core.Syntax$Text(
                              "j"
                            )
                          )
                        ),
                        optInitializer(
                          superc.core.Syntax$Language(
                            "="
                          ),
                          initializer(
                            expression(
                              Conditional(
                                ,
                                superc.core.Syntax$Text(
                                  "5"
                                )
                              )
                            )
                          )
                        ),
                        superc.core.Syntax$Language(
                          ";"
                        )
                      )
                    )
                  )
                ),
                superc.core.Syntax$Language(
                  "}"
                )
              )
            )
          )
        )
      )
    )
  )
)
